cmd/districts.go

```diff
@@ -16,22 +16,42 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
 	"github.com/katta/jabfinder/pkg/cowin"
 	"github.com/spf13/cobra"
+	"log"
 )
 
 // districtsCmd represents the districts command
 var districtsCmd = &cobra.Command{
 	Use:   "districts",
-	Short: "Lists the districts in the given state",
+	Short: "Lists valid districts ( asks for state if not provided )",
 	Run: func(cmd *cobra.Command, args []string) {
 		stateCode, _ := cmd.Flags().GetInt("stateCode")
+
+		if stateCode == 0 {
+			stateCode = getStateInputFromUser()
+		}
+
 		cowin.ListDistricts(stateCode)
 	},
 }
 
+func getStateInputFromUser() int {
```

Comment:
This works, I was thinking using something like https://github.com/manifoldco/promptui which is more interactive but might be an overkill for something as simple as this :)

Reply:
I really liked the Promptui interface. Will move the input gathering to the same in a follow up PR. Thanks Katta! :D